pkg/asset/installconfig/vsphere/validation.go

```diff
@@ -30,6 +35,34 @@ func ValidateForProvisioning(ic *types.InstallConfig) error {
 	}
 
 	allErrs = append(allErrs, validation.ValidateForProvisioning(ic.Platform.VSphere, field.NewPath("platform").Child("vsphere"))...)
+	allErrs = append(allErrs, folderExists(ic, field.NewPath("platform").Child("vsphere").Child("folder"))...)
```

Comment:
hmm.. why did we decide to check at provisioning? would all customers benefit from this validation?

Reply:
this validation requires API access. We don't ask for folder in `create install-config` because it has a default. I _think_ `create cluster` is the only other place we can assume API access.

Reply:
so my understanding is `ValidateForProvisioning` is used to validation before creating cluster.

* install-config validation can also include API calls, we do it for AWS etc.
* before cluster was only for validation that make sense when creating cluster and not much for other targets like manifests, ignition etc.

for example if you give use subnets on AWS, we will validate those even for creating ignition, but we don't check permissions unless create cluster is called.

So my question was, will the vpshere upi customers setting the folder will also benefit from the validation?

Reply:
> So my question was, will the vpshere upi customers setting the folder will also benefit from the validation?

Sorry to not answer your question. The answer is yes. I will check out that AWS validation and move this there.